example/imperative.tsx

```diff
@@ -1,52 +1,39 @@
 import 'react-app-polyfill/ie11';
 import * as React from 'react';
 import { useImperativeFilePicker } from '../src';
+import { PersistentFileAmountLimitValidator } from '../src/validators';
 
 const Imperative = () => {
-  const [
-    openFileSelector,
-    { filesContent, loading, errors, plainFiles, clear, removeFileByIndex, removeFileByReference },
-  ] = useImperativeFilePicker({
-    multiple: true,
-    readAs: 'Text',
-    readFilesContent: true,
-    onFilesSelected: ({ plainFiles, filesContent, errors }) => {
-      // this callback is always called, even if there are errors
-      console.log('onFilesSelected', plainFiles, filesContent, errors);
-    },
-    onFilesRejected: ({ errors }) => {
-      // this callback is called when there were validation errors
-      console.log('onFilesRejected', errors);
-    },
-    onFilesSuccessfulySelected: ({ plainFiles, filesContent }) => {
-      // this callback is called when there were no validation errors
-      console.log('onFilesSuccessfulySelected', plainFiles, filesContent);
-    },
-  });
+  // for imperative file picker, if you want to limit amount of files selected by user, you need to pass persistent validator
+  const validators = React.useMemo(() => [new PersistentFileAmountLimitValidator({ min: 2, max: 3 })], []);
 
-  if (errors.length) {
-    return (
-      <div>
-        <button onClick={() => openFileSelector()}>Something went wrong, retry! </button>
-        <div style={{ display: 'flex', flexDirection: 'column' }}>
-          {console.log(errors)}
-          {Object.entries(errors[0])
-            .filter(([key, value]) => key !== 'name' && value)
-            .map(([key]) => (
-              <div key={key}>{key}</div>
-            ))}
-        </div>
-      </div>
-    );
-  }
+  const { openFilePicker, filesContent, loading, errors, plainFiles, clear, removeFileByIndex, removeFileByReference } =
+    useImperativeFilePicker({
+      multiple: true,
+      readAs: 'Text',
+      readFilesContent: true,
+      validators,
+      onFilesSelected: ({ plainFiles, filesContent, errors }) => {
+        // this callback is always called, even if there are errors
+        console.log('onFilesSelected', plainFiles, filesContent, errors);
+      },
+      onFilesRejected: ({ errors }) => {
+        // this callback is called when there were validation errors
+        console.log('onFilesRejected', errors);
+      },
+      onFilesSuccessfulySelected: ({ plainFiles, filesContent }) => {
```

Comment:
successfully

Reply:
We have a separate PR for that but yeah, I'll fix it here